.changeset/lucky-bulldogs-march.md

```diff
@@ -0,0 +1,20 @@
+---
+'xstate': major
+---
+
+The `internal` property will no longer have effect for transitions on atomic (leaf-node) state nodes. In SCXML, `internal` only applies to complex (compound and parallel) state nodes:
+
+> Determines whether the source state is exited in transitions whose target state is a descendant of the source state. [See 3.13 Selecting and Executing Transitions for details.](https://www.w3.org/TR/scxml/#SelectingTransitions)
+
+```diff
+// ...
+green: {
+  on: {
+    NOTHING: {
```

Comment:
so this only breaks reentering using internal transitions? the description put upfront is quite abstract, would be good to mention concrete scenarios when this thing matters

Reply:
I'll clarify this in the docs, but an internal self-transition in a leaf node is better by not giving a target at all (which is how SCXML represents it) anyway